samza-kv/src/main/java/org/apache/samza/storage/kv/BaseLocalStoreBackedTableProvider.java

```diff
@@ -114,6 +110,30 @@ public Table getTable() {
           SerdeUtils.serialize("Side Inputs Processor", tableSpec.getSideInputsProcessor()));
     }
 
+    // Changelog configuration
+    String enableChangelog = tableSpec.getConfig().get(BaseLocalStoreBackedTableDescriptor.INTERNAL_ENABLE_CHANGELOG);
+    if ("true".equalsIgnoreCase(enableChangelog)) {
```

Comment:
Not sure if this is out of scope for this PR. But, I guess since `TableSpec` is immutable, you can have the config inside TableSpec as a `MapConfig` instead of a `Map<String, String>` so that you can get the type information for free :)

Reply:
Yeah, I'd prefer to take this to another PR.